### PR TITLE
Add gallery for BERTScore

### PR DESCRIPTION
**Pull request by baskrahmer**
## What does this PR do?

Add a gallery for BERTScore

<details>
  <summary>Before submitting</summary>

- [ ] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
📚 Documentation preview 📚: https://torchmetrics--2627.org.readthedocs.build/en/2627/

### CODE REVIEW

File: examples/text/bertscore.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+"""BERTScore
+===============================
+
+BERTScore is a text generation metric to compute the similarity between a generated text and a reference text using a pre-trained BERT model. Instead of relying on exact token matches, BERTScore leverages contextual embeddings to capture the semantic similarity between the texts. This makes BERTScore robust to paraphrasing and word order variations. BERTScore has been shown to correlate well with human judgments and is widely used in evaluating text generation models.
+
+Let's consider a use case in natural language processing where BERTScore is used to evaluate the quality of a text generation model.
```
Comment (Borda) [inline examples/text/bertscore.py:6]: it is good just add an artificial scenario

Comment (SkafteNicki) [inline examples/text/bertscore.py:6]: I am just making something up here
```suggestion
Let's consider a use case in natural language processing where BERTScore is used to evaluate the quality of a text generation model. In this case we are imaging that we are developing a automated news summarization system. The goal is to create concise summaries of news articles that accurately capture the key points of the original articles. To evaluate the performance of your summarization system, you need a metric that can compare the generated summaries to human-written summaries. This is where the BERTScore can be used.
```

File: examples/text/bertscore.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+from transformers import AutoTokenizer, pipeline
+
+pipe = pipeline("text-generation", model="openai-community/gpt2")
+tokenizer = AutoTokenizer.from_pretrained("openai-community/gpt2")
+
+# %%
+# Define the prompt and target texts
+
+prompt = "The quick brown fox jumps"
+target_text = " over the lazy dog."
```
Comment (SkafteNicki) [inline examples/text/bertscore.py:19]: to fit the scenario proposed above 

```suggestion
prompt = "Economic recovery is underway with a 3.5% GDP growth and a decrease in unemployment. Experts forecast continued improvement with boosts from consumer spending and government projects."
target_text = "The economy is recovering, with GDP growth at 3.5% and unemployment at a two-year low. Experts expect this trend to continue due to higher consumer spending and government infrastructure investments."
```

**Review by SkafteNicki — COMMENTED** [review]
I am fine with what is already in this example, but here is a suggestion for a more real life example

**Review by Borda — APPROVED** [review]

**Review by stancld — APPROVED** [review]

**Pull request merged**